h2o-core/src/main/java/water/H2ONode.java

```diff
@@ -390,8 +408,19 @@ synchronized void freeTCPSocket( ByteChannel sock ) {
   // is specifically not any of the above channels.  This channel is limited to
   // messages which are presented in their entirety (not streamed) thus never
   // need another (nested) TCP channel.
-  private transient UDP_TCP_SendThread _sendThread = null; // set notnull if properly interned, and done before first sendMessage
-  public void sendMessage( ByteBuffer bb, byte msg_priority ) { _sendThread.sendMessage(bb,msg_priority); }
+  private transient UDP_TCP_SendThread _sendThread = null; // null if Node was removed from cloud
+  public final void sendMessage(ByteBuffer bb, byte msg_priority) {
+    UDP_TCP_SendThread sendThread = _sendThread;
+    if (sendThread == null) {
+      if (_removed_from_cloud) {
+        Log.warn("Node " + this + " is not active in the cloud anymore but we want to communicate with it." +
+                "Re-opening the communication channel.");
+      }
+      sendThread = startSendThread();
```

Comment:
Just question: If I understand correctly, this is the place where `sendThread` gets initiated for the newly connected clients - ie. if `sentThread == null`

Reply:
Yes, we create the sending threads lazily. This is because we will see (=intern) client nodes that have nothing to do with the cluster. By delaying the initialization to the point when we actually want to send a message, we initialize the sending thread just for the client nodes that are actually part of the cluster.

Reply:
Makes good sense again, small comment in code would help probably why it is done exactly here